docs/Icons.md

```diff
@@ -102,11 +93,36 @@ In some rare cases, you may wish to pass a simple string (e.g., a single letter
 <ComponentName icon={'🍇'} />
 ```
 
-## Image Source
+### Image Source
 
 The icon prop will also support images in various formats, either from local resources or from the web.
 
 ```tsx
 <Component icon={{ uri: 'https://raw.githubusercontent.com/etn-ccis/blui-icons/dev/packages/png/png48/account_settings_black500_48dp.png' }} />
 <Component icon={require('./path/to/image.png')} />
 ```
+
+## Icon as a component
+
+`<Icon>` component is an internal utility component used to render icons inside of other components. It standardizes the implementation of the icon and handles all of the different ways to specify the icon without having to duplicate this logic inside of every component that utilizes icons.
+
+<img width="100%" style="max-width: 300px; display: block;" alt="Icon" src="./images/icon.png">
+
+### Usage
+
+```tsx
+import { Icon } from '@brightlayer-ui/react-native-components';
+
+<Icon source={{ family: 'brightlayer-ui', name: 'device' }} />;
+```
+
+### API
+
+<div style="overflow: auto;">
+
+| Prop Name | Description                        | Type                                   | Required | Default                   |
+| --------- | ---------------------------------- | -------------------------------------- | -------- | ------------------------- |
+| source    | A component to render for the icon | `IconSource`                           | yes      |                           |
```

Comment:
Is "IconSource" a type the same as "Icon Object"'s API? Or are you expecting your user to pass a component, as suggested by the readme?

Reply:
Yes IconSource type is same as icon object

Reply:
You are still saying "A component to render for the icon" here, which suggests to me that you are expecting adopters to pass something that looks like `<IconComponent ... />` for the `source` prop